jdbc/src/main/java/org/talend/components/jdbc/service/JdbcService.java

```
@@ -305,146 +298,256 @@ public static String getSchema(Connection connection) throws SQLException {
+
+                        boolean isKey = keys.contains(columnName);
+
+                        // String defaultValue = tableMetadata.getString("COLUMN_DEF");
+
+                        String columnTypeName = tableMetadata.getString("TYPE_NAME");
+
+                        columnMetadata.put(columnName,
+                                new ColumnMetadata(size, scale, sqlType, nullable, columnName, columnName, null, isKey,
+                                        columnTypeName, null));
+                    }
                 }
-                break;
-            case java.sql.Types.FLOAT:
-            case java.sql.Types.REAL:
-                builder.withEntry(entryBuilder.withType(FLOAT).build());
-                break;
-            case java.sql.Types.DOUBLE:
-                builder.withEntry(entryBuilder.withType(DOUBLE).build());
-                break;
-            case java.sql.Types.BOOLEAN:
-                builder.withEntry(entryBuilder.withType(BOOLEAN).build());
-                break;
-            case java.sql.Types.TIME:
-            case java.sql.Types.DATE:
-            case java.sql.Types.TIMESTAMP:
-                builder.withEntry(entryBuilder.withType(DATETIME).build());
-                break;
-            case java.sql.Types.BINARY:
-            case java.sql.Types.VARBINARY:
-            case java.sql.Types.LONGVARBINARY:
-                builder.withEntry(entryBuilder.withType(BYTES).build());
-                break;
-            case java.sql.Types.BIGINT:
-            case java.sql.Types.DECIMAL:
-            case java.sql.Types.NUMERIC:
-            case java.sql.Types.VARCHAR:
-            case java.sql.Types.LONGVARCHAR:
-            case java.sql.Types.CHAR:
-            default:
-                builder.withEntry(entryBuilder.withType(STRING).build());
-                break;
+            } catch (Exception e) {
+                log.error("[catalog,db schema,table] {} {} {}.", catalog, dbSchema, tableName);
```

here we ignore the exception on purpose, as user's account may not have the rights to access that metadata info!

---

TODO : as we use DatabaseMetadata api:
1. some database have catalog&db schema both
2. some database only have catalog
3. some database only have db schema(not sure).

though we use connection.getSchema to get the db schema, but it depend on database and its jdbc implement. Sometimes, possible, we may get a empty string, or null even that database support db schema or catalog. Image we fetch this on oracle database with lots of db schemas:

```
databaseMetadata.getColumns("mydatabase", "", "a table", xxx)
```
that may be a very slow action as it will search all db schemas, we need to be careful that.

---

test on mysql and oracle, is ok